src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
Ok I see your point. First please make sure this does not break old behavior for end >= 0

We need more discussion on this since it breaks consistency with numpy @tqchen @pluskid @antinucleon